### PR TITLE
Include main job in worfklow

### PR DESCRIPTION
**Pull request by d2lam**
User can't have `main` in workflow, but it will still show up by default.

### CODE REVIEW

File: lib/phase/functional.js [hunk 1]
```diff
@@ -145,10 +142,19 @@ function validateJobSchema(doc) {
...
-    });
+    const workflow = Hoek.reach(doc, 'workflow');
+
+    // if not defined by user, get default workflow
+    if (!workflow) {
+        return getDefaultWorkflow(doc.jobs);
+    }
+
+    // if defined by user, prepend with 'main'
+    const newWorkflow = workflow.slice(0);
```
Comment (stjohnjohnson) [inline lib/phase/functional.js]: use `clone` instead

File: lib/phase/functional.js [hunk 2]
```diff
@@ -145,10 +142,19 @@ function validateJobSchema(doc) {
...
 function generateWorkflow(doc) {
-    // Default to the list of jobs
-    return Hoek.reach(doc, 'workflow', {
-        default: getDefaultWorkflow(doc.jobs)
-    });
+    const workflow = Hoek.reach(doc, 'workflow');
+
+    // if not defined by user, get default workflow
+    if (!workflow) {
+        return getDefaultWorkflow(doc.jobs);
```
Comment (stjohnjohnson) [inline lib/phase/functional.js]: Replace with `Object.keys(doc.jobs)` and remove `getDefaultWorkflow`

File: lib/phase/functional.js [hunk 3]
```diff
@@ -145,10 +142,19 @@ function validateJobSchema(doc) {
...
+    }
+
+    // if defined by user, prepend with 'main'
+    const newWorkflow = workflow.slice(0);
+
+    newWorkflow.unshift('main');
+
+    return newWorkflow;
```
Comment (stjohnjohnson) [inline lib/phase/functional.js]: `return ['main'].concat(workflow);`

File: lib/phase/functional.js [hunk 4]
```diff
@@ -160,8 +166,10 @@ function generateWorkflow(doc) {
  * @return {Array}               List of errors
  */
 function validateWorkflow(doc) {
-    // Validate that main is not included
-    if (doc.workflow.indexOf('main') > -1) {
```
Comment (stjohnjohnson) [inline lib/phase/functional.js]: Why not ensure indexOf === 0

**Comment by stjohnjohnson** [comment]
Sure, why not! 👍 

magic

**Comment by FenrirUnbound** [comment]
👍 

Looks good to me
bqly37q

**Pull request merged**